pkg/chart/repo/handlers.go

```diff
@@ -0,0 +1,51 @@
+package repo
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	helmchart "k8s.io/helm/pkg/proto/hapi/chart"
+	"k8s.io/helm/pkg/repo"
+
+	shipper "github.com/bookingcom/shipper/pkg/apis/shipper/v1alpha1"
+)
+
+type ChartVersionResolver func(*shipper.Chart) (*repo.ChartVersion, error)
+
+type ChartFetcher func(*shipper.Chart) (*helmchart.Chart, error)
+
+var ResolveChartVersionFunc = func(c *Catalog) ChartVersionResolver {
+	return func(chartspec *shipper.Chart) (*repo.ChartVersion, error) {
+		repo, err := c.CreateRepoIfNotExist(chartspec.RepoURL)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create repo[%s] object: %s", chartspec.RepoURL, err)
+		}
+
+		if err := repo.RefreshIndex(); err != nil {
+			glog.Warningf("failed to refresh repo[%s] index: %s", chartspec.RepoURL, err)
+		}
+
+		cv, err := repo.ResolveVersion(chartspec.Name, chartspec.Version)
+		if err != nil {
+			return nil, fmt.Errorf("failed to resolve chart version: %v", err)
+		}
+
+		return cv, nil
+	}
+}
+
+var FetchChartFunc = func(c *Catalog) ChartFetcher {
```

Comment:
Out of curiosity, why are this and `ResolveChartVersionFunc` assigned to vars instead of being a normal func? It makes me think there's something special to them, but I was not able to figure out why.

Reply:
Good catch! It's an artefact: initially I redefined these functions in the tests, but later on transitioned to module attribute injection. To be fixed.